### PR TITLE
Replace transactions getter interfaces with storage.entities.Transaction.get - Closes #2621

### LINKED ISSUE #2621: Replace transactions getter interfaces with storage.entities.Transaction.get
Replace `db.transactions.list` and  `modules.transactions[.shared].get*` with `storage.entieis.Transaction.get`

### PR DESCRIPTION
**Pull request by nazarhussain**
### What was the problem?
Dependency of transactions getter functions from account modules were used in API. Making it a strict dependency for API.

### How did I fix it?
For all persisted transactions data, fetched the code directly from the database.

### How to test it?
Run transactions functional tests.

### Review checklist

* The PR resolves #2621
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: api/controllers/transactions.js [hunk 1]
```diff
@@ -54,60 +70,56 @@ TransactionsController.getTransactions = function(context, next) {
...
-		fromHeight: params.height.value,
-		toHeight: params.height.value,
-		fromTimestamp: params.fromTimestamp.value,
-		toTimestamp: params.toTimestamp.value,
-		minAmount: params.minAmount.value,
-		maxAmount: params.maxAmount.value,
-		data: params.data.value,
+		height: params.height.value,
+		timestamp_gte: params.fromTimestamp.value,
+		timestamp_let: params.toTimestamp.value,
```
Comment (pablitovicente) [inline api/controllers/transactions.js]:
```suggestion
		timestamp_lte: params.toTimestamp.value,
```

File: api/controllers/transactions.js [hunk 2]
```diff
@@ -54,60 +70,56 @@ TransactionsController.getTransactions = function(context, next) {
...
 	let filters = {
-		senderIdOrRecipientId: params.senderIdOrRecipientId.value,
 		id: params.id.value,
 		blockId: params.blockId.value,
 		recipientId: params.recipientId.value,
 		recipientPublicKey: params.recipientPublicKey.value,
 		senderId: params.senderId.value,
 		senderPublicKey: params.senderPublicKey.value,
 		type: params.type.value,
-		fromHeight: params.height.value,
```
Comment (pablitovicente) [inline api/controllers/transactions.js]: Why is this filter removed?

File: api/controllers/transactions.js [hunk 3]
```diff
@@ -54,60 +70,56 @@ TransactionsController.getTransactions = function(context, next) {
...
-		toHeight: params.height.value,
```
Comment (pablitovicente) [inline api/controllers/transactions.js]: Why is this filter removed?

File: api/controllers/transactions.js (hunk 2 above)
Comment (nazarhussain) [reply api/controllers/transactions.js]: Just mapped the param to direct filter of `height`

File: storage/entities/transaction.js [hunk 4]
```diff
@@ -144,6 +145,15 @@ const assetAttributesMap = {
 	7: ['asset.outTransfer.dappId', ' asset.outTransfer.transactionId'],
 };
 
+// eslint-disable-next-line no-unused-vars
+const stringToByteOnlyInsert = (value, mode, alias, fieldName) => {
```
Comment (lsilvs) [inline storage/entities/transaction.js]: Why not add this to `utils/inputSerializers.js`?

File: api/controllers/transactions.js (hunk 1 above)
Comment (MaciejBaj) [reply api/controllers/transactions.js]: ☝️

File: test/unit/storage/entities/transaction.js [hunk 5]
```diff
@@ -182,8 +324,35 @@ describe('Transaction', () => {
...
+				id: transaction.id,
+			});
+
+			expect(result).to.be.a('number');
+			return expect(result).to.be.eql(1);
+		});
+
+		it('should resolve with zero if matching record not found', async () => {
+			const result = await storage.entities.Transaction.count({
+				id: '1234',
```
Comment (MaciejBaj) [inline test/unit/storage/entities/transaction.js]: Use the constants expressing of what the value represents, like `const NON_EXISTENT_ID = '1234';`

File: storage/entities/transaction.js [hunk 6]
```diff
@@ -181,17 +191,27 @@ class Transaction extends BaseEntity {
...
 			},
-			stringToByte
+			stringToByteOnlyInsert
+		);
+		this.addField(
+			'recipientPublicKey',
+			'string',
+			{
+				filter: filterTypes.TEXT,
+				format: 'publicKey',
```
[the remark this answers is not in the document]
Comment (nazarhussain) [reply storage/entities/transaction.js]: I placed it to generate dynamic `json-schema` to validate the data input. But we can implement it later.

File: storage/entities/transaction.js [hunk 7]
```diff
@@ -25,11 +25,12 @@ const BaseEntity = require('./base_entity');
  * @typedef {Object} BasicTransaction
  * @property {string} id
  * @property {string} blockId
- * @property {Integer} [blockHeight]
+ * @property {Integer} [height]
```
[the remark this answers is not in the document]
Comment (nazarhussain) [reply storage/entities/transaction.js]: Its kept for consistency.

File: storage/entities/transaction.js [hunk 8]
```diff
@@ -520,13 +547,28 @@ class Transaction extends BaseEntity {
...
+
+		// To have deterministic pagination add extra sorting
+		if (parsedOptions.sort) {
+			parsedOptions.sort = _.flatten([
+				parsedOptions.sort,
+				't_rowId:asc',
+			]).filter(Boolean);
+		} else {
+			parsedOptions.sort = ['t_rowId:asc'];
+		}
```
[the remark this answers is not in the document]
Comment (nazarhussain) [reply storage/entities/transaction.js]: Need to handle the case when user passes sort as a string.

File: storage/entities/transaction.js (hunk 4 above)
Comment (nazarhussain) [reply storage/entities/transaction.js]: This is a special case for transaction only. That `SELECT` query return a `string` while `INSERT` query expect a Buffer.

**Pull request merged**